interview-questions.md

```diff
@@ -8,50 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is executing a command through a predefined array until a condition is met. The last portion, until a condition is met, is crucial to remember so you don't have a function stopping at a condition much longer or farther than the intended one. Iteration can do something for each value in an array. For example, iteration could take an array of numbers and return the array with all values multiplied by three. 
 
-Researched answer:
+Researched answer: Iteration occurs when we make a decision about every value in an array. Iteration is highlighted in for loops and higher-order functions with .forEach, .map, and .filter. Iteration is dynamic. Iteration can manipulate data and return an array of the same length, or extract only the specified data required to meet the function's conditions. Iteration is extremely useful when a programmer wants to evaluate every element/value in an array or execute a repetetive command.
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: The difference between a class and an object is a class is a blueprint of an object and an object determines the behavior of data. Classes are static as they have the same exact skeleton for each class, while an object can be dynamic via functions inside the object, called methods.
 
-Researched answer:
+Researched answer: A class is a function that contains data and behavior. An object is an instance of a class. Classes hold properties and act as a shell or blueprint for an object. Classes have certain keywords required to create and access data from within a class. For example, the constructor creates the object inside the class, "this" is used to refer to an object inside the class, and "new" is used when creating a new object (if class = Coffee, then const lattee = new Coffee; latte would now have access to the properties inside of Coffee). A class is written in PascalCase, whereas, objects are written in camelCase. Objects are considered the crossroads between data and behavior because they can hold functions internally, called methods. Objects contain key:value pairs, where key is the data type of symbol and value can be any valid JavaScript data type. An example of this would look like: const person = [name: "Tabi", phone: "9499031938"]; key would be name and phone and value would be "Tabi" and "9499031938". 
 
-6. STRETCH: What is hoisting in JavaScript?
+6. STRETCH: What is hoisting in JavaScript? 
 
-Your answer:
+Your answer: Hoisting in JavaScript is attaching another command onto an invoked fucntion when it is logged.
 
-Researched answer:
+Researched answer: Hoisting allows a programmer to enable a function before it is recorded in their code. Hoisitng happens when the function declaration or variables are transfered to the top of a programmer's scope before the fucntion is invoked.
```

Comment:
Really great work in this section!!  All of your answers are really well-defined and I really like the examples where you provide examples!